arch/arm/core/aarch64/reset.S

```diff
@@ -63,6 +66,56 @@ SECTION_SUBSEC_FUNC(TEXT,_reset_section,__start)
 	msr	spsel, #0
 
 	ldr	x0, =(z_interrupt_stacks)
+
+#define IS_POWER_OF_TWO(X) ((X) && (((X) & ((X) - 1)) == 0))
```

Comment:
Might be good to put `IS_POWER_OF_TWO` in include/sys/util.h, already has a C inline func version.

Reply:
I could do that.

Some questions around how you do things "around here" that I haven't really seen answered elsewhere, likely somewhat silly ones:

1. If I did so, would I keep the title as arch: arm64 seeing as I modified shared code? Or?
2. Add `IS_POWER_OF_TWO` as a separate commit (i.e. rewrite history of this PR such that the first commit adds `IS_POWER_OF_TWO` and the second uses it)? Or as part of the same commit? Or as a different PR?
3. Does Zephyr care about tests passing on intermediate commits in a single PR? (Some projects do, some don't).

Reply:
> * If I did so, would I keep the title as arch: arm64 seeing as I modified shared code? Or?
> 
For something trivial no need to change the title.

> * Add `IS_POWER_OF_TWO` as a separate commit (i.e. rewrite history of this PR such that the first commit adds `IS_POWER_OF_TWO` and the second uses it)? Or as part of the same commit? Or as a different PR?
> 

Again, for something trivial its up to you to decide.  It makes sense to keep in the same PR since it shows the code that is using it as well.

> * Does Zephyr care about tests passing on intermediate commits in a single PR? (Some projects do, some don't).

We do care about this, but aren't strictly enforcing it.

Reply:
Thanks for the answers!

Reply:
I may have overkilled this somewhat. Could you please take another look?